### PR TITLE
Some Fixings & Reenabling some tests

### PR DESCRIPTION
**Pull request by nupplaphil**
I reenabled some tests and catched two bugs:
- `array_reverse()` didn't do anything because the output wasn't mapped to the corresponding lists
- Twitter `/statuses` endpoint didn't work, because we never used the `list_id` and statically used `1`

I'm not done yet, but at least some tests are working again :-)

btw. the test coverage dropped from 15% to 12% the last weeks :-/ .. We do need to move the API tests as well

### CODE REVIEW

File: src/Module/Api/Mastodon/Accounts/VerifyCredentials.php [hunk 1]
```diff
@@ -52,6 +52,6 @@ protected function rawContent(array $request = [])
 
 		// @todo Support the source property,
 		$account = DI::mstdnAccount()->createFromContactId($cdata['user'], $uid);
-		System::jsonExit($account);
+		$this->response->exit('account', ['account' => $account->toArray()]);
```
Comment (annando) [inline src/Module/Api/Mastodon/Accounts/VerifyCredentials.php]: Mastodon endpoints only support a JSON output. So wouldn't it be better having just a `jsonExit` function in the `response` class?

File: tests/src/Module/Api/Mastodon/Accounts/StatusesTest.php [hunk 2]
```diff
@@ -11,6 +14,8 @@ class StatusesTest extends ApiTest
 	 */
 	public function testApiStatusShowWithJson()
 	{
+		self::markTestIncomplete('Needs Stastuses to not set header during call (like at BaseApi::setLinkHeader');
```
Comment (annando) [inline tests/src/Module/Api/Mastodon/Accounts/StatusesTest.php]: `Statuses`

File: src/Module/Api/Mastodon/Accounts/VerifyCredentials.php (hunk 1 above)
Comment (nupplaphil) [reply src/Module/Api/Mastodon/Accounts/VerifyCredentials.php]: Currently, it's JSON per default, but I will enforce it per parameter

Comment (annando) [reply src/Module/Api/Mastodon/Accounts/VerifyCredentials.php]: What I meant was that we arbitrary add parameters here (see `'account'`) that are only used when not exporting via JSON. This is not elegant.

**Comment by nupplaphil** [comment]
OK, I'll leave it for now because we're in the RC phase.

But -> I already have a PR in the pipeline which allows us to throw HTTPExceptions (and MastodonAPI exceptions) instead of the `exit()`s, so there's no need for custom `return` points anymore inside Modules :-). But I stashed it for the develop phase again :)

File: src/Module/Api/Twitter/Blocks/Lists.php [hunk 3]
```diff
@@ -87,6 +87,6 @@ protected function rawContent(array $request = [])
 
 		self::setLinkHeader();
 
-		System::jsonExit($return);
+		$this->response->exit('lists', ['lists' => $return]);
```
Comment (MrPetovan) [inline src/Module/Api/Twitter/Blocks/Lists.php:90]:
```suggestion
		$this->response->exitWithJson($return);
```

File: src/Module/Api/Twitter/Followers/Lists.php [hunk 4]
```diff
@@ -92,6 +92,6 @@ protected function rawContent(array $request = [])
 
 		self::setLinkHeader();
 
-		System::jsonExit($return);
+		$this->response->exit('lists', ['lists' => $return]);
```
Comment (MrPetovan) [inline src/Module/Api/Twitter/Followers/Lists.php:95]:
```suggestion
		$this->response->exitWithJson($return);
```

File: src/Module/Api/Twitter/Friends/Lists.php [hunk 5]
```diff
@@ -92,6 +92,6 @@ protected function rawContent(array $request = [])
 
 		self::setLinkHeader();
 
-		System::jsonExit($return);
+		$this->response->exit('lists', ['lists' => $return]);
```
Comment (MrPetovan) [inline src/Module/Api/Twitter/Friends/Lists.php:95]:
```suggestion
		$this->response->exitWithJson($return);
```

File: src/Module/Api/Twitter/Friendships/Incoming.php [hunk 6]
```diff
@@ -86,6 +86,6 @@ protected function rawContent(array $request = [])
 
 		self::setLinkHeader();
 
-		System::jsonExit($return);
+		$this->response->exit('incoming', ['incoming' => $return]);
```
Comment (MrPetovan) [inline src/Module/Api/Twitter/Friendships/Incoming.php:89]:
```suggestion
		$this->response->exitWithJson($return);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Api/Twitter/Followers/Lists.php (hunk 4 above)
Comment (annando) [reply src/Module/Api/Twitter/Followers/Lists.php:95]: The Twitter endpoints should support XML.

**Review by annando — APPROVED** [review]

File: src/Module/Api/Twitter/Blocks/Lists.php (hunk 3 above)
Comment (annando) [reply src/Module/Api/Twitter/Blocks/Lists.php:90]: The Twitter endpoints should support XML.

File: src/Module/Api/Twitter/Friends/Lists.php (hunk 5 above)
Comment (annando) [reply src/Module/Api/Twitter/Friends/Lists.php:95]: The Twitter endpoints should support XML.

File: src/Module/Api/Twitter/Friendships/Incoming.php (hunk 6 above)
Comment (annando) [reply src/Module/Api/Twitter/Friendships/Incoming.php:89]: The Twitter endpoints should support XML.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**